rpc/src/client.rs

```diff
@@ -1,31 +1,45 @@
 //! Tendermint RPC client
 
-use bytes::buf::ext::BufExt;
-use hyper::header;
+use tendermint::{
+    abci::{self, Transaction},
+    block::Height,
+    evidence::Evidence,
+    net, Genesis,
+};
 
-use tendermint::abci::{self, Transaction};
-use tendermint::block::Height;
-use tendermint::evidence::Evidence;
-use tendermint::net;
-use tendermint::Genesis;
-
-use crate::{endpoint::*, Error, Request, Response};
+use crate::{
+    client::{
+        subscription::SubscriptionManager,
+        transport::{http_ws::HttpWsTransport, Transport},
+    },
+    endpoint::*,
+    Error, Request, Response,
+};
 
 pub mod event_listener;
+pub mod subscription;
+pub mod transport;
+
+#[cfg(test)]
+pub mod testing;
 
 /// Tendermint RPC client.
 ///
 /// Presently supports JSONRPC via HTTP.
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 pub struct Client {
-    /// Address of the RPC server
-    address: net::Address,
+    transport: Box<dyn Transport>,
```

Comment:
Why do we need to `Box` this instead of having it as a [type parameter](https://doc.rust-lang.org/reference/types/parameters.html)?

Reply:
I considered the implications of having it as a type parameter as opposed to a trait object and I'm not sure which of the two approaches provides the best ergonomics in the long run. Of course, using trait objects (which seem to be used pretty extensively in the Light Client module) results in simpler higher-level interfaces, they come with their own challenges.

I can certainly experiment with the type parameter-based approach instead. It would certainly help to be able to use generics in the `Transport` trait, for example.